logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamImpl.java

```diff
@@ -331,17 +337,23 @@ public void addFailureListener(final FailureListener failureListener) {
 
   @Override
   public void onFailure() {
-    if (failureListener != null) {
-      failureListener.onFailure();
-    }
-    closeAsync();
+    actor.run(
```

Comment:
Why we need to schedule a job here now?

Reply:
This can be called externally by LogStorageAppender when it fails.